R/peaks-functions.R

```diff
@@ -90,6 +90,54 @@ NULL
     x[which(between(x[, "intensity"], intensity)), , drop = FALSE]
 }
 
+#' @description
+#'
+#' Filter the spectrum keeping only peaks that match the provided m/z value(s).
+#'
+#' @inheritParams .peaks_remove
+#'
+#' @param mz `numeric` with the m/z values to match against. This needs to be
+#'     a sorted `numeric` (has to be checked upstream).
+#'
+#' @param tolerance `numeric` with the tolerance. Can be of length 1 or equal
+#'     length `mz`.
+#'
+#' @param ppm `numeric` with the ppm. Can be of length 1 or equal length `mz`.
+#'
+#' @author Johannes Rainer
+#'
+#' @importFrom MsCoreUtils closest
+#'
+#' @noRd
+.peaks_filter_mz_value <- function(x, spectrumMsLevel, mz = numeric(),
+                                   tolerance = 0, ppm = 20,
+                                   msLevel = spectrumMsLevel, ...) {
+    if (!spectrumMsLevel %in% msLevel || !length(x))
+        return(x)
+    idx <- closest(mz, x[, "mz"], tolerance = tolerance, ppm = ppm,
+                   duplicates = "closest")
+    x[idx[!is.na(idx)], , drop = FALSE]
```

Comment:
That's exactly what `common` does:
```suggestion
    keep <- common(mz, x[, "mz"], tolerance = tolerance, ppm = ppm,
                   duplicates = "closest")
    x[keep, , drop = FALSE]
```

Reply:
Done in https://github.com/rformassspectrometry/Spectra/commit/a80db29a4601e43b176513e4a364f75e1af14a23

Reply:
Actually, I had to revert this. Calling `common` in that way returns a `logical` of length 1. If we call it with `common(x[, "mz"], mz, ...)` it would return a logical with the correct length, but then `tolerance` and `ppm` can no longer be of length `= length(mz)`.

Reply:
Would like to get your OK on reverting that back @sgibb